### PR TITLE
rbac: deprecate mixer rbac adapter

### PR DESCRIPTION
**Pull request by yangminzhu**
It's replaced by local RBAC in 1.1 release.

#6377

### CODE REVIEW

**Comment by mandarjog** [comment]
Is the local version fully functionally equivalent ? What about generates attributes ?

**Comment by yangminzhu** [comment]
@mandarjog 
The local RBAC currently doesn't support TCP protocol. I have finished the RBAC network filter in Envoy, will soon finish Istio change to support it.

The other difference is local RBAC only supports [a subset of attributes](https://istio.io/docs/reference/config/authorization/constraints-and-properties/). Basically only those trusted attributes (e.g. principal from cert) could be used as identities in the RBAC policy.

File: samples/bookinfo/platform/kube/istio-rbac-enable.yaml [hunk 1]
```diff
@@ -1,41 +0,0 @@
-apiVersion: "config.istio.io/v1alpha2"
-kind: authorization
```
Comment (rshriram) [inline samples/bookinfo/platform/kube/istio-rbac-enable.yaml]: why are you removing examples? isn't it referred in the docs?

Comment (yangminzhu) [reply samples/bookinfo/platform/kube/istio-rbac-enable.yaml]: These are old samples for mixer based rbac, the docs now referrers to the new samples in `samples/bookinfo/platform/kube/rbac`

**Comment by liminw** [comment]
@yangminzhu As we discussed, let's hold on this change until we complete TCP level support.

**Comment by yangminzhu** [comment]
@rshriram @geeknoid @mandarjog I have rebased this PR on release-1.1 branch, PTAL, thanks!

**Comment by geeknoid** [comment]
What are yout thoughts on compat?

In theory, we shipped this adapter in 1.0 and so we need to support it for two releases, meaning we wouldn't actually delete it until 1.3. Additionally, we would need to provide a migration story for customers that might have used the adapter. Do you think this level of compatibility is not necessary?

**Comment by yangminzhu** [comment]
@geeknoid 
I think the adapter is shipped before 1.0, I remember it's added in 0.7 or 0.8? @liminw to confirm.

The RBAC policy is backward compatible, user could just use their existing RBAC policies without making any changes. We could write a blog or simply include a note in the release note to let the user to read the new [doc](https://preliminary.istio.io/docs/concepts/security/#authorization)?

**Comment by geeknoid** [comment]
If I have config that's using the adapter, and we delete the adapter, then my config will no longer work, right? Mixer will complain it doesn't know the adapter and fail to init correctly.

We have committed to compatibility for 2 LTS releases. That means customers can upgrade and stuff keeps working.

Perhaps we need to keep the adapter present, but have it be a NOP except for outputting some log entry stating the adapter is no longer supported. We keep this for 1.1 and 1.2, and delete the adapter completely in 1.3.

WDYT?

**Comment by yangminzhu** [comment]
@geeknoid Sure, I will update this PR to make it a dummy adapter for now, and remove it completely in 1.3.

**Comment by yangminzhu** [comment]
@geeknoid I have updated the PR to make the rbac adapter to always return an error message, PTAL. Thanks.

File: mixer/adapter/rbac/rbac.go [hunk 2]
```diff
@@ -15,198 +15,38 @@
...
-
-// adapter.Handler#Close
-func (h *handler) Close() error {
-	h.closing <- true
-	close(h.closing)
-
-	<-h.done
-	h.store.Stop()
-	return nil
+	return nil, env.Logger().Errorf(prompt)
```
Comment (geeknoid) [inline mixer/adapter/rbac/rbac.go]: I think you need to return success here, otherwise the operator will get errors after upgrade.

**Comment by geeknoid** [comment]
I think you should also remove the // $XXX comments from config.proto such that the docs for the adapter aren't published to istio.io.

File: mixer/adapter/rbac/rbac.go (hunk 2 above)
Comment (yangminzhu) [reply mixer/adapter/rbac/rbac.go]: But I thought that's our intention? To notify the operator to upgrade to proxy based RBAC?
When I tried this locally, it doesn't crash mixs but only output an error message in the log.

**Comment by geeknoid** [comment]
OK, I thought Mixer would just bail. If Mixer just returns an error, than I think this looks good.

**Pull request merged**